core/Cargo.toml

```diff
@@ -22,11 +22,16 @@ libc = "0.2.98"
 log = "0.4.14"
 parking_lot = "0.11.1"
 pin-project = "1.0.7"
+reqwest = { version = "0.11.4", default-features = false, features = ["rustls-tls", "stream", "gzip", "brotli"] }
+rustls = "0.19.0"
+rustls-native-certs = "0.5.0"
```

Comment:
I want to double-check: are other maintainers on board with taking on these dependencies?

Reply:
We already have those dependencies, but it's definitely not ok to add anything related to HTTP and/or TLS to `core/`. These changes should only happen in `extensions/`, `runtime/` and `cli/` directories.

Reply:
The rustls-native-certs is a new one, one of the comments in this PR adds some information for reviewers to review the dependency. The other two were already added.

This stuff was moved into core because a previous review stated that there was a lot of duplicate code (there was) and so to consolidate it I made one function and had the multiple places in code call those new shared functions.

I'll look to move it to runtime, I have no opinion one way or another it just seemed like core was a library which each of the callers was already referencing.

Reply:
https://github.com/denoland/deno/pull/11491#pullrequestreview-713068769

Reply:
@justinmchase moving to `runtime/` won't help as you need to use those crates in various crates in `extensions/` directory. 

I don't want you to do busy work trying guess where a "good" place for them might be. I will give your PR full review over the weekend and we can figure out a proper place for them then.

Reply:
Okay, now I get what's going on in this PR; I suggest to add new crate in `extensions/` directory - let's call it `deno_tls` that will contain TLS utilities used by other crates. 

This will be useful for features implemented in https://github.com/denoland/deno/pull/11324

Once that's done it should be more or less done (provided that tests are passing)